native/src/main/java/io/ballerina/stdlib/http/transport/contractimpl/sender/states/http2/SendingRstFrame.java

```diff
@@ -0,0 +1,113 @@
+import io.ballerina.stdlib.http.transport.contractimpl.common.states.Http2MessageStateContext;
+import io.ballerina.stdlib.http.transport.contractimpl.sender.http2.Http2ClientChannel;
+import io.ballerina.stdlib.http.transport.contractimpl.sender.http2.Http2TargetHandler;
+import io.ballerina.stdlib.http.transport.contractimpl.sender.http2.OutboundMsgHolder;
+import io.ballerina.stdlib.http.transport.message.Http2DataFrame;
+import io.ballerina.stdlib.http.transport.message.Http2HeadersFrame;
+import io.ballerina.stdlib.http.transport.message.Http2PushPromise;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.handler.codec.http.HttpContent;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.netty.handler.codec.http2.Http2ConnectionEncoder;
+import io.netty.handler.codec.http2.Http2Error;
+import io.netty.handler.codec.http2.Http2Exception;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static io.ballerina.stdlib.http.transport.contract.Constants.IDLE_TIMEOUT_TRIGGERED_WHILE_WRITING_OUTBOUND_REQUEST_HEADERS;
+import static io.ballerina.stdlib.http.transport.contract.Constants.REMOTE_SERVER_CLOSED_WHILE_WRITING_OUTBOUND_REQUEST_HEADERS;
+
+/**
+ * A state to reset the stream in the middle of communication.
+ */
+public class SendingRstFrame implements SenderState {
+
+    private static final Logger LOG = LoggerFactory.getLogger(SendingRstFrame.class);
+
+    private final Http2TargetHandler http2TargetHandler;
+    private final Http2MessageStateContext http2MessageStateContext;
+    private final OutboundMsgHolder outboundMsgHolder;
+    private final Http2ClientChannel http2ClientChannel;
+    private final Http2ConnectionEncoder encoder;
+    private int streamId;
+    private final Http2TargetHandler.Http2RequestWriter http2RequestWriter;
+
+    public SendingRstFrame(Http2TargetHandler http2TargetHandler,
+                           Http2TargetHandler.Http2RequestWriter http2RequestWriter) {
+        this.http2TargetHandler = http2TargetHandler;
+        this.http2MessageStateContext = http2RequestWriter.getHttp2MessageStateContext();
+        this.outboundMsgHolder = http2RequestWriter.getOutboundMsgHolder();
+        this.http2ClientChannel = http2TargetHandler.getHttp2ClientChannel();
+        this.encoder = http2TargetHandler.getEncoder();
+        this.streamId = http2RequestWriter.getStreamId();
+        this.http2RequestWriter = http2RequestWriter;
+    }
+
+    @Override
+    public void writeOutboundRequestHeaders(ChannelHandlerContext ctx, HttpContent httpContent) throws Http2Exception {
+        LOG.warn("writeOutboundRequestHeaders is not a dependant action of this state");
+    }
+
+    @Override
+    public void writeOutboundRequestBody(ChannelHandlerContext ctx, HttpContent httpContent,
+                                         Http2MessageStateContext http2MessageStateContext) throws Http2Exception {
+        writeRstStream(ctx);
```

Comment:
Call `resetStream()` from here

Reply:
Added the changes in https://github.com/ballerina-platform/module-ballerina-http/pull/659/commits/287830336fb2b11641e3547b9ee16b050e076d95